runner/runner.go

```diff
@@ -0,0 +1,198 @@
+package runner
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/consul-template/manager"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+//go:embed ansible
+var ansibleFS embed.FS
+
+const (
+	AnsibleMain = "ansible/main.yaml"
+)
+
+type Runner struct {
+	cfg            Config
+	ctx            context.Context
+	ctxCancel      context.CancelFunc
+	templateRunner *manager.Runner
+}
+
+type Config struct {
+	AraServer              string
+	ConsulAddr             string
+	Interval               time.Duration
+	AnsibleFolder          string
+	ConsulTemplateLogLevel string
+}
+
+func NewWithConfig(cfg Config) (*Runner, error) {
+	templateRunner, err := NewTemplateRunner(&cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create the consul template runner")
+	}
+
+	ctx, ctxCancel := context.WithCancel(context.Background())
+
+	runner := &Runner{
+		cfg:            cfg,
+		ctx:            ctx,
+		ctxCancel:      ctxCancel,
+		templateRunner: templateRunner,
+	}
+
+	return runner, nil
+}
+
+func DefaultConfig() (Config, error) {
+	return Config{
+		AraServer:              "http://127.0.0.1:8000",
+		ConsulAddr:             "127.0.0.1:8500",
+		Interval:               5 * time.Minute,
+		AnsibleFolder:          "/usr/etc/trento",
```

Comment:
Move this to `/var/lib/trento` (check for temporary folders)

Reply:
Changed in: https://github.com/trento-project/trento/pull/150/commits/6a4281fc34ca8eee7d4a811a1bf9f042d5fa1423